File: 733-flood-fill/flood-fill.cpp
class Solution {
public:
    void ans(vector<vector<int>>& image, int sr, int sc, int color, int curr){

    }
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int color) {
        if (image[sr][sc] == color){
            return image;
        }
        int farRight = image[0].size() - 1;
        int farDown = image.size() -1;
        int first = image[sr][sc];
        image[sr][sc] = color;
        if(sc > 0 && image[sr][sc-1] == first){
            floodFill(image, sr,sc-1,color);     
        }
        if(sc < farRight && image[sr][sc+1] == first){
            floodFill(image, sr,sc+1,color);
        }
        if(sr < farDown && image[sr+1][sc] == first){
            floodFill(image, sr+1,sc,color);
        }
        if(sr > 0 && image[sr-1][sc] == first){
            floodFill(image, sr-1,sc,color);
        }
        // cout<<image.size();
        return image;
    }
};

File: 733-flood-fill/flood-fill.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, color: int) -> List[List[int]]:
        if image[sr][sc] == color:
            return image 
        starting_pixel = image[sr][sc]
        image[sr][sc] = color
        
        if sc-1 >=0 and image[sr][sc-1] == starting_pixel:
            image = self.floodFill(image, sr, sc-1,color)
            image[sr][sc-1] = color
        if sc+1 <= len(image[0])-1 and image[sr][sc+1] == starting_pixel: 
            image = self.floodFill(image, sr, sc+1, color)
            image[sr][sc+1] = color
        if sr-1 >= 0 and image[sr-1][sc] == starting_pixel:
            image = self.floodFill(image, sr-1,sc, color)
            image[sr-1][sc] = color
        if sr+1 <= len(image)-1 and image[sr+1][sc] == starting_pixel:
            image = self.floodFill(image, sr+1, sc, color)
            image[sr+1][sc] = color
        
        return image